src/store/epics/__tests__/wallets.test.ts

```diff
@@ -1,73 +1,173 @@
-import { unlockWalletFailure, unlockWalletRequest, unlockWalletSuccess } from '@actions'
-import { unlockWallet$, walletLockFailure$ } from '../wallets'
+import {
+  tooManyFailedAttempts,
+  unlockWalletFailure,
+  unlockWalletFailureAlert,
+  unlockWalletRequest,
+  unlockWalletSuccess
+} from '@actions'
+import {
+  tooManyFailedAttempts$,
+  unlockWallet$,
+  unlockWalletFailureAlert$,
+  walletLockFailure$
+} from '../wallets'
 
 import { epicTest } from './helpers'
 
 describe('wallets epic', () => {
   describe('unlock failure', () => {
     it('throws failure alert on failure', async () => {
       const generalFailureAlert = jest.fn(() => Promise.resolve())
+
+      await epicTest({
+        epic: unlockWalletFailureAlert$,
+        inputActions: [unlockWalletFailureAlert()],
+        expectedActions: [],
+        dependencies: { generalFailureAlert },
+      })
+
+      expect(generalFailureAlert).toHaveBeenCalled()
+    })
+
+    it('throws a failure alert on too many failed attempts', async () => {
+      const generalFailureAlert = jest.fn(() => Promise.resolve())
+
       await epicTest({
-        epic: walletLockFailure$,
-        inputActions: [unlockWalletFailure()],
+        epic: tooManyFailedAttempts$,
+        inputActions: [tooManyFailedAttempts(new Date())],
         expectedActions: [],
         dependencies: { generalFailureAlert },
       })
+
       expect(generalFailureAlert).toHaveBeenCalled()
     })
   })
+})
 
-  describe('unlock wallet request', () => {
-    it('sends success action', async () => {
-      const decryptedPrivateKey = 'success'
-      const publicKey = 'publicKey'
-      const password = 'password'
+describe('unlock wallet request', () => {
+  const dateObj = Date
 
-      const decryptPrivateKey = jest.fn(() => decryptedPrivateKey)
+  beforeEach(() => {
+    const dateStub = jest.fn(() => Date)
```

Comment:
This is a bit of a code smell to me, I'd prefer to have some function that we can call to give us the timestamp we are expecting, rather than a side-effect inside the epic that we can't reproduce without doing something like overriding the whole global Date object. Maybe it's not that big a deal, but seeing it's a side effect I'd prefer to put the call to Date.now/new Date() into a dependency for the epic to use. We can then just give it a fixed value in the tests.